### ISSUE TITLE
NWInt work like NWFloat

### ISSUE THREAD

**Issue by HavannaFR** [Facepunch/garrysmod-issues#4409]
### Details
The NWInt function does not work properly because it should round the number as NW2Int does, but it does not, it works exactly like NWFloat, it can store numbers that are not integers.

### Steps to reproduce
```
] lua_run Entity(1):SetNWInt("test", 0.1)
> Entity(1):SetNWInt("test", 0.1)...
] lua_run print(Entity(1):GetNWInt("test", 0.1))
> print(Entity(1):GetNWInt("test", 0.1))...
0.1
] lua_run Entity(1):SetNW2Int("test", 0.1)
> Entity(1):SetNW2Int("test", 0.1)...
] lua_run print(Entity(1):GetNW2Int("test", 0.1))
> print(Entity(1):GetNW2Int("test", 0.1))...
0
```

It can be problematic for addons who want to use NWInt directly to round store and round numbers without needing to do so before.

**Comment by robotboy655** [comment]
Yes, and unfortunately people have been using it this way for years and that's how they expect it to work, so changing it would potentially break shit.

**Comment by HavannaFR** [comment]
It's sad because the transition from MW to NW2 when it takes place will cause same problem anyway :/

**Comment by mcNuggets1** [comment]
Never saw a single addon doing it this way and even I didn't know of this before now.

**Comment by HavannaFR** [comment]
> Never saw a single addon doing it this way and even I didn't know of this before now.

For exemple, VcMod use NWInt for storing no integerer number for vehicle health, pretty bad ...

**Comment by HavannaFR** [comment]
> Never saw a single addon doing it this way and even I didn't know of this before now.

For exemple, VcMod use NWInt for storing no integerer number for vehicle health, pretty bad ...

**Comment by mcNuggets1** [comment]
They do? I always thought it was clamped.
I use the dev functions of it anyway, so I couldn't know.

Well, if that's the case, I agree with Rubat, that the behaviour shouldn't be changed.
Maybe document it on the wiki page.

**Comment by HavannaFR** [comment]
Yeah, I reported the "error" to freeman, idk if he's gonna fix that or not ...

But yes I'm agree with you, change can break lot of addons, but NW2 gonna replace NW some day no ?
If the answer is yes the result is same.

**Comment by Kefta** [comment]
I think an alternative should be added when NW2 is ported to NW if compatibility is in mind, such as ``*NWInteger`` for a real int. Also ``*NWUnsigned`` would be really useful.
